app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,9 @@
+class CompaniesController < ApplicationController
+
+  #before_action :authenticate_user!
+
+  def create
+    @company = Company.create
```

Comment:
a Company should `belong_to` a user, we're preferring simplicity over performance